src/auth/lobby.ts

```diff
@@ -100,6 +104,58 @@ export const isUsernameAvailable = async (username: string): Promise<boolean> =>
   return user.isUsernameAvailable(username)
 }
 
+export const createChannel = (options: ChannelOptions): Promise<Channel> => {
+  return channel.createChannel(options)
+}
+
+export const checkCapability = async (username: string): Promise<boolean> => {
+  const readKey = await storage.getItem("readKey")
+  if (!readKey) return false
+
+  const didFromDNS = await did.root(username)
+  const maybeUcan: string | null = await storage.getItem("ucan")
+
+  if (maybeUcan) {
+    const rootIssuerDid = token.rootIssuer(maybeUcan)
+    const decodedUcan = token.decode(maybeUcan)
+    const { ptc } = decodedUcan.payload
+
+    return didFromDNS === rootIssuerDid && ptc === "SUPER_USER"
+  } else {
+    const rootDid = await did.write()
+
+    return didFromDNS === rootDid
+  }
+}
+
+export const delegateAccount = async (username: string, audience: string): Promise<Record<string, unknown>> => {
+    const readKey = await storage.getItem("readKey")
+    const proof = await storage.getItem("ucan") as string
+
+    const u = await token.build({
+      audience,
+      issuer: await did.write(),
+      lifetimeInSeconds: 60 * 60 * 24 * 30 * 12 * 1000, // 1000 years
+      potency: "SUPER_USER",
+      proof,
+  
+      // TODO: UCAN v0.7.0
+      // proofs: [ await localforage.getItem("ucan") ]
+    })
+  
+    return { readKey, token: token.encode(u) }
+}
+
+export const linkDevice = async (data: Record<string, unknown>): Promise<void> => {
```

Comment:
Could we require the `token` and `readKey` params specifically here, so it's clear from looking at the docs what the function expects?

Reply:
I think we need to type to match the type in the implementation

https://github.com/fission-suite/webnative/blob/787db781ff4061736fc8c1f6225e68be7de06984/src/auth/implementation/types.ts#L14

and we want to keep that type general to suit different account semantics. But maybe we could use generics somehow to form an explicit contract between `delegateAccount` and `linkDevice`? 🤔 

I'm not sure how we would do this, but will ponder it. Any ideas or thoughts on other ways we could approach this?

Reply:
Ah gotcha 👍 I would just add some documentation then. Not worried about the type checking or anything here.